### PR TITLE
fix(documentDB): prevent field addition from flattener to exceed 30 fields to meet with DocumentDB limitation

### PR DESCRIPTION
**Pull request by SteveBunlon**
### General

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [x] Test manually the implemented changes
- [x] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [x] Consider the security impact of the changes made

### CODE REVIEW

File: packages/datasource-mongoose/src/utils/pipeline/reparent.ts [hunk 1]
```diff
@@ -74,11 +74,22 @@ export default class ReparentGenerator {
...
-          { $project: Object.fromEntries(asFields.map(f => [f, 0])) },
-        ]
-      : [];
+    if (asFields.length === 0) {
+      return [];
+    }
+
+    const unflattenResults = [];
+    const addFields = asFields.map(f => [f.replace(/\./g, '@@@'), `$${f}`]);
+    // DocumentDB limits the addFields stage to 30 fields.
```
Comment (Thenkei) [inline packages/datasource-mongoose/src/utils/pipeline/reparent.ts:83]: Is this only a DocumentDB limitation ? Does it have an impact on performance for non DocumentDB DB ?

Comment (SteveBunlon) [reply packages/datasource-mongoose/src/utils/pipeline/reparent.ts:83]: [This](https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html#mongo-apis-dababase-aggregation) might also be handy

Comment (SteveBunlon) [reply packages/datasource-mongoose/src/utils/pipeline/reparent.ts:83]: On top of that, I would say that the query system itself is under-performant. I don't why, we do addFields for the flattened fields, where we don't project them. 

If I take a quick look at the query being generated, I have no filters whatsoever, and the system adds my flattened attribute as fields, but doesn't project them in the end 🤔

Comment (SteveBunlon) [reply packages/datasource-mongoose/src/utils/pipeline/reparent.ts:83]: In a more global approach, I tend to think that the query system is under performant. If I don't need any of my flattened fields to be used in a sort or filter, their's no reason to add them using `addFields`. Aliasing them in the $project pipeline would be enough like so : $project: {my@@@nested: $my.nested}. I would rather change this in the regard of performance.

Comment (SteveBunlon) [reply packages/datasource-mongoose/src/utils/pipeline/reparent.ts:83]: I can deep dive into performances if you want me to

**Review by Thenkei — APPROVED** [review]
Let's merge it like this and we gonna rework the reparent using only `$project`.

**Pull request merged**